### PR TITLE
Blog content - WIP

### PR DESCRIPTION
**Pull request by drsm79**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
+_scripts/makepost.py 'Your post title' your-post-category
+```
+
+If the category is new, a stub file will be created for it in the top level directory. These stubs are used to build a category index. You will need to commit this file as well as the post file to git.
+
+You can run `jekyll serve` in the top directory of the site to serve a local copy of the site & verify markdown or any edits to styling, layouts or includes at http://localhost:4000. The built artefacts will be in `_site`, which can be useful for debugging.
+
+## Editing the layout & appearance
+
+Pages are given a layout (in `_layouts`) which may use one or more include (from `_includes`). The sites CSS is in `style.css`
```
Comment (alfinkel) [inline README.md]:
```suggestion
Pages are given a layout (in `_layouts`) which may use one or more include (from `_includes`). The site's CSS is in `style.css`.
```

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Framework is smart so fetchers & checks don't need to be
+
+The reason for building Auditree is to improve the scaling of our compliance efforts. We wanted engineers to be able to dip in, write a `unittest` based check & go back to their normal activities. To enable this, we built the framework, with the aim of it holding the "smarts" and the fetchers/checks only containing the logic and sufficient intelligence required to collect & evaluate evidence.
+
+We've exposed this functionality as [decorators and context managers][evidence], as well as [`Fetcher`][fetcher] and [`Check`][check] classes. Using these mean people can write their code and not worry about evidence TTL, where files reside in the locker or managing git as the framework takes care of these functions for you.
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md]: Initially I read `...Using these mean people...` as though the people were mean :) .  Perhaps change it to `...Using these mean that people...`?

Comment (alfinkel) [reply _posts/2020-08-13-Goals_general_architecture.md]: - The link to "decorators and context managers" being a link to the evidence code module doesn't seem helpful.   Maybe reword things a bit so that you can provide links to [fetcher design principles evidence validation](https://complianceascode.github.io/auditree-framework/design-principles.html#evidence-validation) and [fetcher design principles evidence validation](https://complianceascode.github.io/auditree-framework/design-principles.html#id2) which both provide some details on the decorators and context managers for both fetchers and checks.
- Similar comment for the fetcher and check links.  Seems like it would be more helpful to provide links to the fetcher and check design principles.  https://complianceascode.github.io/auditree-framework/design-principles.html#compliance-fetchers and https://complianceascode.github.io/auditree-framework/design-principles.html#compliance-checks.
- Lastly, any reason you chose to back tic and capitalize "Fetcher" and "Check"?  The rendered markdown looks odd.

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## Runner agnostic
+
+Initially we ran Auditree on Travis; we have a private instance & it required config but no new infrastructure be deployed. The environment & install is simple & portable between different runners, and we've ensured the framework is agnostic of it's running environment - we've run it in Jenkins, from cron, in Tekton. All that's required is the ability to verify that the runner has not been tampered with.
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md]: Suggest changing:

>Initially we ran Auditree on Travis; we have a private instance & it required config but no new infrastructure be deployed.

to 

> Initially we ran Auditree on Travis; we have a private instance which required some configuration but no new infrastructure had to be deployed.

Comment (alfinkel) [reply _posts/2020-08-13-Goals_general_architecture.md]: Suggest changing:

> ...and we've ensured the framework is agnostic of it's running environment - we've run it in Jenkins, from cron, in Tekton

to

> ...and we've ensured the framework is agnostic of its running environment - we've run it in Jenkins, from cron, in Tekton as well as in Travis.

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+One thing we baked in early on is that each run is a new install. This means we can rectify issues in fetcher/check code or configuration and have the next run pick up the fix, without intervention. It also means that the ony state shared between invocations is managed via the locker, preventing any "ghosts in the machine" from changing evidence.
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md]: - "ony" -> "only"
- This is the first mention of the locker in the post, maybe add some supporting text on the locker or a link to https://complianceascode.github.io/auditree-framework/design-principles.html#evidence-locker or both.

**Review by alfinkel — CHANGES_REQUESTED** [review]

File: _posts/2020-08-13-Goals_general_architecture.md (hunk 2 above)
Comment (drsm79) [reply _posts/2020-08-13-Goals_general_architecture.md]: I didn't want to link to line numbers, since they'll be wrong immediately after the next update. Good call onlinking to the design principles. I wanted them to be code e.g. a `Fetcher` not a fetcher.

File: _posts/2020-08-13-Goals_general_architecture.md (hunk 4 above)
Comment (drsm79) [reply _posts/2020-08-13-Goals_general_architecture.md]: I'll add a paragraph or two for the locker

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+## Framework is smart so fetchers & checks don't need to be
+
+The reason for building Auditree is to improve the scaling of our compliance efforts. We wanted engineers to be able to dip in, write a `unittest` based check & go back to their normal activities. To enable this, we built the framework, with the aim of it holding the "smarts" and the fetchers/checks only containing the logic and sufficient intelligence required to collect & evaluate evidence.
+
+We've exposed this functionality as decorators and context managers that can be used to [write][] and [read][] evidence, as well as [`Fetcher`][fetcher] and [`Check`][check] classes. Using these mean that people can write their code and not worry about evidence TTL, where files reside in the locker or managing git as the framework takes care of these functions for you.
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md]: Because we haven't mentioned TTL or locker until now...

```suggestion
We've exposed this functionality as decorators and context managers that can be used to [write][] and [read][] evidence, as well as [`Fetcher`][fetcher] and [`Check`][check] classes. Using these mean that people can write their code and not worry about evidence TTL (time to live), where files reside in the evidence locker (git repository) or managing git as the framework takes care of these functions for you.
```

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## The Locker
+
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md:20]: We still need this bit and then I think we're good to go.

**Review by alfinkel — CHANGES_REQUESTED** [review]

File: _posts/2020-08-13-Goals_general_architecture.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+Auditree focuses on evidence. We wanted to use a long term store which:
+
+- Met familiarity goals of the project as a whole
+- Maintained record integrity & provided evidence of tampering
+- Could deliver a change history via a defined API
+- Had suitable access & permission controls
+- Was able to support a range of data formats (text based or binary)
+- was able to meet the retention requirements associated with the durations of audits
```
Comment (alfinkel) [inline _posts/2020-08-13-Goals_general_architecture.md]:
```suggestion
- Was able to meet the retention requirements associated with the durations of audits
```

**Review by alfinkel — APPROVED** [review]
One tweak
